gobblin-core/src/main/java/org/apache/gobblin/source/PartitionAwareFileRetrieverUtils.java

```diff
@@ -52,4 +60,29 @@ public static Duration getLeadTimeDurationFromConfig(State state) {
 
     return new Duration(leadTime * leadTimeGranularity.getUnitMilliseconds());
   }
+
+  /**
+   * Calculates the lookback time duration based on the provided lookback time string.
+   *
+   * @param lookBackTime the lookback time string, which should include a numeric value followed by a time unit character.
+   *                     For example, "5d" for 5 days or "10h" for 10 hours.
+   * @return an {@link Optional} containing the {@link Duration} if the lookback time is valid, or
+   *         an empty {@link Optional} if the lookback time is invalid or cannot be parsed.
+   */
+  public static Optional<Duration> getLookbackTimeDuration(String lookBackTime) {
```

Comment:
although I am a big proponent of `Optional`, in this case it loses the specific errors, putting them in the log rather than "returning" them to the caller, where they might be presented in a meaningful end-user error message (rather than logs, which are for developers).

since string parsing like this regularly requires error handling, it seems reasonable and basically expected to throw an exception.  you can avow in javadoc that the `Duration` return value is never `null`

Reply:
Updated to throw IOException